osu.Game/Database/BeatmapDatabase.cs

```diff
@@ -188,7 +219,7 @@ public void Import(string path)
                 }
             }
 
-            Import(new[] { beatmapSet });
+            return beatmapSet;
         }
 
         public void Import(IEnumerable<BeatmapSetInfo> beatmapSets)

```

Comment:
seems as though this should not be public, as the only way we can guarantee things are in a good state is if this is called by another method above inside of `BeatmapDatabase`. Might even work better to combine this with the `Import(paths)` method above.